service/history/workflow/task_refresher.go

```diff
@@ -217,8 +216,14 @@ func (r *TaskRefresherImpl) refreshTasksForWorkflowClose(
 	executionState := mutableState.GetExecutionState()
 
 	if executionState.Status != enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING {
+		closeEvent, err := mutableState.GetCompletionEvent(ctx)
```

Comment:
You can call GetWorkflowCloseTime() to get the close time

Reply:
Plz check my response to the other comments.